src/fsharp/FSharp.Core.Unittests/FSharp.Core/Microsoft.FSharp.Control/AsyncModule.fs

```diff
@@ -191,6 +199,41 @@ type AsyncModule() =
 
         for _i = 1 to 50 do test()
 
+
+    [<Test>]
+    member this.``Async.AwaitWaitHandle does not leak memory`` () =
```

Comment:
This test is too expensive. This 1 case by itself takes around 5 min to run, which is somewhere around 70%+ of the overall suite execution time. :hourglass: 

In F# 3.0 I get at least 3 or 4 failures here if I do just 100 iterations.  Reducing to that level cuts the time down to something reasonable.